hydrant/models/patient.py

```diff
@@ -39,22 +49,44 @@ def as_fhir(self):
         results.update(self._fields)
         return results
 
-    def as_upsert_entry(self):
+    def as_upsert_entry(self, target_system=None):
         """Generate FHIR for inclusion in transaction bundle
 
         Transaction bundles need search and method details for
         FHIR server to perform requested task.
 
+        :param target_system: define to perform lookup for existing
         :returns: JSON snippet to include in transaction bundle
         """
         results = {}
         results['resource'] = self.as_fhir()
-        # TODO look up Patient to see if identifier is found, and if so, PUT
-        # with method=POST, a new will be generated each time.
+        method = 'POST'
+
+        # FHIR spec: 'birthDate'; HAPI search: 'birthdate'
+        patient_url = (
+            f'Patient?family={self._fields["name"]["family"]}&'
+            f'given={self._fields["name"]["given"][0]}&'
+            f'birthdate={self._fields["birthDate"]}')
+
+        # Round-trip to see if this represents a new or existing Patient
+        if target_system:
+            response = requests.get('/'.join((target_system, patient_url)))
```

Comment:
Having `requests` build the querystring params is slightly cleaner
```suggestion
        patient_url = "Patient"
        search_params = {
            "family": self._fields["name"]["family"],
            "given": self._fields["name"]["given"][0],
            "birthdate": self._fields["birthDate"],
        }

        # Round-trip to see if this represents a new or existing Patient
        if target_system:
            response = requests.get('/'.join((target_system, patient_url)), params=search_params)
```